src/main/java/com/tester/processor/TestProcessor.java

```diff
@@ -1,267 +1,45 @@
-      sb.append(method.getName());
-      sb.append(method.getDesc());
-      sb.append("\n");
-      sb.append("Line coverage: ");
-      sb.append(method.getLineCounter().getCoveredRatio() * 100);
-      sb.append("%");
-      sb.append("\n");
-      sb.append("Branch coverage: ");
-      sb.append(method.getBranchCounter().getCoveredRatio() * 100);
-      sb.append("%");
-      sb.append("\n\n");
-    }
-    sb.append("Here are the suspicious lines in the program not covered by any test in the " +
-            "test suite:\n");
-    Map<Integer, Boolean> lineCoverageStatus = summary.getLineCoverageStatus();
-    StringBuilder susLinesSb = new StringBuilder();
-    String filePath = getFilePath(MAIN, type, programFileName);
-    for (int l : suspiciousLines.get(type)) {
-      if (lineCoverageStatus.getOrDefault(l, false)) {
-        continue;
-      }
-      String programLine = getProgramLine(l, filePath);
-      susLinesSb.append("Line ");
-      susLinesSb.append(l);
-      susLinesSb.append(": ");
-      susLinesSb.append(programLine);
-      susLinesSb.append("\n");
-    }
-    if (susLinesSb.length() == 0) {
-      sb.append("None");
-    } else {
-      sb.append(susLinesSb);
+    @Override
+    String getFilePath(String dir, String type, String fileName) {
+        return BaseProcessor.getSRC() + dir + BaseProcessor.getEXAMPLES() + "/" + type + "/" + fileName;
     }
-    return sb.toString();
-  }
 
-  private void writeToFile(String type, String code) {
-    byte[] codeBytes = code.getBytes();
-    String packageName = getPackageName(type);
-    byte[] packageNameBytes = packageName.getBytes();
-    String filePath = getFilePath(TEST, type, testFileName);
-    try {
-      if (Files.exists(Paths.get(filePath))) {
-        Files.delete(Paths.get(filePath));
-      }
-      FileChannel channel = FileChannel.open(Paths.get(filePath), StandardOpenOption.CREATE,
-              StandardOpenOption.APPEND);
-      channel.write(ByteBuffer.wrap(packageNameBytes));
-      channel.write(ByteBuffer.wrap(codeBytes));
-      channel.force(true);
-    } catch (IOException e) {
-      e.printStackTrace();
+    @Override
+    String getClassPath(String type, String fileName) {
+        return BaseProcessor.getTargetDir() + BaseProcessor.getEXAMPLES() + "/" + type + "/" + getBaseName(fileName) + BaseProcessor.getCLASS();
     }
-  }
 
-  private String getProgramLine(int lineNum, String filePath) {
-    int currLineNum = 1;
-    try {
-      BufferedReader reader = new BufferedReader(new FileReader(filePath));
-      String line = reader.readLine();
-      while (line != null && currLineNum < lineNum) {
-        line = reader.readLine();
-        currLineNum++;
-      }
-      return line;
-    } catch (IOException e) {
-      e.printStackTrace();
-      return "";
+    @Override
+    String getQualifiedClassName(String type) {
+        return BaseProcessor.getEXAMPLES() + "." + type + "." + getBaseName(super.getTestFileName());
```

Comment:
instead of `super.getTestFileName()` just make the fields in the base class protected